extendedcopy.go

```diff
@@ -104,6 +105,36 @@ func ExtendedCopyGraph(ctx context.Context, src content.GraphStorage, dst conten
 	return nil
 }
 
+// FilterArtifactType will configure opts.FindPredecessors to filter the predecessors
+// whose annotation matches a given regex pattern. The regex syntax supported
+// is RE2. Reference: https://github.com/google/re2/wiki/Syntax.
+func (opts *ExtendedCopyGraphOptions) FilterAnnotation(key string, pattern string) {
```

Comment:
There is a concern on the performance for filtering multiple keys. For example, if we want to filter two keys, we should call `FindSuccessors` only once instead of twice.

Reply:
Filtering multiple times will not cause extra costs.